discovery/sync_manager.go

```diff
@@ -375,7 +375,21 @@ func (m *SyncManager) syncerHandler() {
 		// Our HistoricalSyncTicker has ticked, so we'll randomly select
 		// a peer and force a historical sync with them.
 		case <-m.cfg.HistoricalSyncTicker.Ticks():
-			m.forceHistoricalSync()
+			s := m.forceHistoricalSync()
+
+			// If we've already performed our initial historical
+			// sync, then we have nothing left to do.
+			if m.IsGraphSynced() {
+				continue
+			}
+
+			// Otherwise, we'll track the peer we've performed a
```

Comment:
interesting. So this is really an existing bug, but it wouldn't really matter that the `initialHistoricalSyncCompleted` was never set?

Reply:
Exactly. It would matter as it could prevent us from transitioning to active syncers.